libr/include/r_util/r_rbtree.h

```diff
@@ -25,7 +25,7 @@ typedef struct r_rb_node_t {
 
 typedef RBNode* RBTree;
 
-typedef int (*RBComparator)(const void *incoming, const RBNode *in_tree);
+typedef int (*RBComparator)(const void *incoming, const RBNode *in_tree, void *user);		//here needs to be a void *user
```

Comment:
imho the RBComparator should take the RBTree as first arg, and this void* user should be inside the RBTree struct